test/Graphics/Transform.cpp

```diff
@@ -6,17 +6,22 @@
 #include <GraphicsUtil.hpp>
 #include <vector>
 
-namespace std
+// Use StringMaker to avoid opening namespace std
+namespace doctest
 {
-std::ostream& operator<<(std::ostream& out, const std::vector<float>& vector)
+template <>
+struct StringMaker<std::vector<float>>
 {
-    out << "{ ";
-    for (size_t i = 0; i + 1 < vector.size(); ++i)
-        out << vector[i] << ", ";
-    out << vector.back() << " }";
-    return out;
-}
-} // namespace std
+    static String convert(const std::vector<float>& vector)
+    {
+        doctest::String out = "{ ";
+        for (size_t i = 0; i + 1 < vector.size(); ++i)
```

Comment:
Great attention to detail, avoiding `i < vector.size() - 1`! 👍

Reply:
I actually did that for style reasons but now I'm realizing that if `size()` returns `0` then _ouch_. I've never thought of that failure mode before!

Reply:
As a side note, an empty vector would fail two lines later, with `vector.back()`. Fortunately we do not print empty vectors.

Reply:
That's ironic, it was _precisely_ the style that made me think "why the hell does he not compare the loop variable directly" 😅 

Do you think it's worth a comment regarding underflow, or is it clear enough?

Reply:
@kimci86 Good catch!

Maybe `if (vector.empty())` special case + early return? Then we could also use _normal_™ style 😛 

If the assumption is "we never print empty vectors", we could also do `assert(!vector.empty())` -- better than UB and potentially flaky tests for sure.

Reply:
```cpp
        if (vector.empty())
            return "{ }";
        doctest::String out = "{ ";
        for (size_t i = 0; i + 1 < vector.size(); ++i)
            out += toString(vector[i]) + ", ";
        out += toString(vector.back()) + " }";
        return out;
```
A guard clause seems like the easiest option if we want to catch empty vectors

Reply:
Seems like we're currently printing no empty vectors so we can get away with an assert and just wait for this to maybe become a problem later. I think that's what I'd prefer for now just to be safe and without us spending too much time on this.